src/sentry/api/endpoints/organization_details.py

```diff
@@ -954,7 +950,45 @@ def put(self, request: Request, organization) -> Response:
             with transaction.atomic(router.db_for_write(Organization)):
                 organization, changed_data = serializer.save()
 
-            if "targetSampleRate" in changed_data:
+            if (
+                organization.get_option("sentry:sampling_mode")
+                == DynamicSamplingMode.ORGANIZATION.value
+            ) and "targetSampleRate" in changed_data:
+                with transaction.atomic(router.db_for_write(ProjectOption)):
+                    organization.update_option(
+                        "sentry:target_sample_rate", request.data["targetSampleRate"]
+                    )
+
+            if "samplingMode" in changed_data and request.access.has_scope("org:write"):
+                if (
+                    organization.get_option("sentry:sampling_mode")
+                    == DynamicSamplingMode.PROJECT.value
+                ):
+                    with transaction.atomic(router.db_for_write(ProjectOption)):
+                        for project in organization.project_set.all():
+                            current_rate, is_fallback = get_boost_low_volume_projects_sample_rate(
+                                org_id=organization.id,
+                                project_id=project.id,
+                                error_sample_rate_fallback=None,
+                            )
+                            if current_rate:
+                                project.update_option("sentry:sampling_rate", current_rate)
+
+                        organization.delete_option("sentry:target_sample_rate")
+
+                if (
+                    organization.get_option("sentry:sampling_mode")
+                    == DynamicSamplingMode.ORGANIZATION.value
+                ):
+                    with transaction.atomic(router.db_for_write(ProjectOption)):
+                        if "targetSampleRate" in changed_data:
+                            organization.update_option(
+                                "sentry:target_sample_rate", request.data["targetSampleRate"]
+                            )
+
+                        for project in organization.project_set.all():
+                            project.delete_option("sentry:sampling_rate")
```

Comment:
```suggestion
                        ProjectOption.objects.filter(project__in=organization.project_set.all()).filter(key="sentry:sampling_rate").delete()
```

I am not sure if there is some downside that I don't see, but instead of N queries we could run only 1

Reply:
Another way to do this is via a join, which even avoids us to load all the projects.

```suggestion
                        ProjectOption.objects.filter(project__organization_id=organization.id, key="sentry:target_sample_rate").delete()
```